src/test/java/io/supertokens/test/accountlinking/GetUserByIdTest.java

```diff
@@ -0,0 +1,141 @@
+                null, "123456");
+        return Passwordless.consumeCode(main, code.deviceId, code.deviceIdHash,
+                code.userInputCode, null).user;
+    }
+
+    @Test
+    public void testAllLoginMethods() throws Exception {
+        String[] args = {"../"};
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{
+                        EE_FEATURES.ACCOUNT_LINKING, EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        AuthRecipeUserInfo user1 = createEmailPasswordUser(process.getProcess(), "test@example.com", "password1");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user2 = createThirdPartyUser(process.getProcess(), "google", "userid1", "test@example.com");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user3 = createPasswordlessUserWithEmail(process.getProcess(), "test@example.com");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user4 = createPasswordlessUserWithPhone(process.getProcess(), "+919876543210");
+
+        assertFalse(user1.isPrimaryUser);
+        assertFalse(user2.isPrimaryUser);
+        assertFalse(user3.isPrimaryUser);
+        assertFalse(user4.isPrimaryUser);
+
+        assertEquals(user1, AuthRecipe.getUserById(process.getProcess(), user1.id));
+        assertEquals(user2, AuthRecipe.getUserById(process.getProcess(), user2.id));
+        assertEquals(user3, AuthRecipe.getUserById(process.getProcess(), user3.id));
+        assertEquals(user4, AuthRecipe.getUserById(process.getProcess(), user4.id));
+
+        AuthRecipeUserInfo primaryUser = AuthRecipe.createPrimaryUser(process.getProcess(), user4.id).user;
+        AuthRecipe.linkAccounts(process.getProcess(), user3.id, primaryUser.id);
+        AuthRecipe.linkAccounts(process.getProcess(), user2.id, primaryUser.id);
+        AuthRecipe.linkAccounts(process.getProcess(), user1.id, primaryUser.id);
+
+        for (String userId : new String[]{user1.id, user2.id, user3.id, user4.id}) {
+            AuthRecipeUserInfo result = AuthRecipe.getUserById(process.getProcess(), userId);
+            assertTrue(result.isPrimaryUser);
+
+            assertEquals(4, result.loginMethods.length);
+            assertEquals(user1.loginMethods[0], result.loginMethods[0]);
+            assertEquals(user2.loginMethods[0], result.loginMethods[1]);
+            assertEquals(user3.loginMethods[0], result.loginMethods[2]);
+            assertEquals(user4.loginMethods[0], result.loginMethods[3]);
+
+            assertEquals(user1.timeJoined, result.timeJoined);
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
```

Comment:
- need to test that this function returns null if user does not exist
- need to test that the order of login methods is sorted in the correct way (if you remove the sorting logic, the test should fail - make sure of that)

Reply:
Done